### PR TITLE
Create New Onboarding

### PR DESCRIPTION
**Pull request by pnor**
And google sign in

### CODE REVIEW

File: Fitness/Controllers/EndOnboardingLetterViewController.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+//  EndOnboardingLetterViewController.swift
+//  Fitness
+//
+//  Created by Phillip OReggio on 3/21/19.
+//  Copyright © 2019 Keivan Shahida. All rights reserved.
+//
+
+import UIKit
+
+class EndOnboardingLetterViewController: UIViewController {
```
Comment (jmf373) [inline Fitness/Controllers/EndOnboardingLetterViewController.swift]: Is there any reason we have this file? If we don't use it we should just delete it

File: Fitness/Controllers/OnboardingGymsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    // MARK: - INITIALIZATION
+    var titleLabel: UILabel!
+    var nextButton: UIButton!
+    var nextButtonArrow: UIImageView!
+    var backButton: UIButton!
+    var backButtonArrow: UIImageView!
+    
+    // TableView
+    var gymsTableView: UITableView!
+    let reuseIdentifier = "gymsTableView"
```
Comment (jmf373) [inline Fitness/Controllers/OnboardingGymsViewController.swift]: Maybe add this as a static field of the cell to which it refers (it's done this way with the other reusable cells/headers we have)? This way having several cell types is simpler and it is more clear which cell type the reuse identifier refers to

File: Fitness/Controllers/OnboardingGymsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                self.gymNames = namesArray
+                self.gymsTableView.reloadData()
+            }
+        }
+
+        setUpConstraints()
+    }
+
+    func setUpConstraints() {
+        titleLabel.snp.makeConstraints { (make) in
```
Comment (jmf373) [inline Fitness/Controllers/OnboardingGymsViewController.swift]: Nitpicky but remove the parentheses around `make`

File: Fitness/Controllers/OnboardingGymsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        
+        backButton = UIButton()
+        backButton.clipsToBounds = false
+        backButton.layer.cornerRadius = nextButtonSize / 2
+        backButton.backgroundColor = .fitnessWhite
+        backButton.layer.borderColor = UIColor.fitnessMediumGrey.cgColor
+        backButton.layer.borderWidth = buttonBorderSize
+        backButton.addTarget(self, action: #selector(goBackAView), for: .touchDown)
+        backButton.isEnabled = false
+        backButtonArrow = UIImageView(image: UIImage(named: "arrow"))
```
Comment (jmf373) [inline Fitness/Controllers/OnboardingGymsViewController.swift]: See what design thinks about this, but in the habit view controller we used a lighter grey back button without the yellow, not sure if they would want that here too for consistency

File: Fitness/Controllers/OnboardingGymsViewController.swift [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    
+    //MARK: Button Actions
+    @objc func goToNextView() {
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else {
+            print("FAILED...")
+            return
+        }
+        let defaults = UserDefaults.standard
+        defaults.set(true, forKey: Identifiers.hasSeenOnboarding)
+        appDelegate.window?.rootViewController = TabBarController()
```
Comment (jmf373) [inline Fitness/Controllers/OnboardingGymsViewController.swift]: Shouldn't we be going to the `OnboardingLetterViewController` at this point?

File: Fitness/Controllers/OnboardingLoginViewController.swift [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+            make.top.equalTo(view.safeAreaLayoutGuide).inset(edgePadding)
+            make.leading.trailing.equalToSuperview().inset(edgePadding)
+        }
+        
+        signUpLabel.snp.makeConstraints { (make) in
+            make.top.equalTo(view.safeAreaLayoutGuide).inset(titleToContentPadding)
+            make.leading.trailing.equalToSuperview().inset(edgePadding)
+        }
+        
+//        emailField.snp.makeConstraints { (make) in
```
Comment (jmf373) [inline Fitness/Controllers/OnboardingLoginViewController.swift]: If you don't think we'll use this code (or the previous that initializes it) it in the future, delete it!

File: Fitness/Controllers/OnboardingMessageViewController.swift [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+//  Fitness
+//
+//  Created by Phillip OReggio on 3/14/19.
+//  Copyright © 2019 Keivan Shahida. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+class OnboardingMessageViewController: UIViewController {
```
Comment (jmf373) [inline Fitness/Controllers/OnboardingMessageViewController.swift]: Any reason for this file?

File: Fitness/Controllers/OnboardingGymsViewController.swift (hunk 2 above)
Comment (pnor) [reply Fitness/Controllers/OnboardingGymsViewController.swift]: Ok, will change!

File: Fitness/Controllers/EndOnboardingLetterViewController.swift (hunk 1 above)
Comment (pnor) [reply Fitness/Controllers/EndOnboardingLetterViewController.swift]: We didn't have a finalized Onboarding Letter view at the time, so I was making what was there at the time (meant to delete this before, oops)

**Pull request merged**